### DISCUSSION TITLE
The requirements are inconsistent with the description.

### DISCUSSION THREAD

**Discussion by Enchante503** [erew123/alltalk_tts#312] [Q&A] [answered]
ALL Systems - General requirements
Python version 3.11.9
PyTorch version 2.2.1 with CUDA 12.1

It says this, but when I actually run it, I get an error.

```
 [WARNING]  Please specify the CUTLASS repo directory as environment variable $CUTLASS_PATH
 [WARNING]  sparse_attn requires a torch version >= 1.5 and < 2.0 but detected 2.2
 [WARNING]  using untested triton version (2.2.0), only 1.0.0 is known to be compatible
```
 
```
 requirements_standalone.txt
 nvidia-cublas-cu11>=11.11.3.6
nvidia-cudnn-cu11>=9.1.1.17
```

This is also not CUDA12, it is different from the description.

Please show me the developer's pip list.

**Comment by erew123** [comment, marked as answer]
Hi @Enchante503 

The Cutlass message is related to DeepSpeed and its to do with how it compiles and current issues with it. Microsoft are working on this. Please see here as an example conformation from Logan @ Microsoft https://github.com/microsoft/DeepSpeed/issues/5470. I expect when Microsoft start finally releasing full wheel files in the next month or two, these should be resolvable. It is possible to specify the cutlass path to negate this issue if you wish. There is reference to the error message existing/being ignorable on the Linux DeepSpeed releases https://github.com/erew123/alltalk_tts/releases/tag/DeepSpeed-14.2-Linux

Re nvidia-cublas-cu11 and nvidia-cudnn-cu11, these are parts of the Nvidia CUDA Development Toolkit https://developer.nvidia.com/cuda-toolkit and not related to the Python/Pytorch CUDA version the system is running. They were mainly in relation to a few aspects of Coqui's scripts and also FasterWhisper. FasterWhisper has moved to Cublas and Cudnn v12 around 2-3 months ago https://github.com/SYSTRAN/faster-whisper/commit/3d1de60ef3ce7d34f7c0ae6547f8a616aa060ac2 And I have been waiting on an update to the Coqui scripts, which should happen some time in the next month. When thats completed, I will look to migrate the cublas and cdnn to version 12, pending no other conflicts.

If you wish to look through the requirements files they are here https://github.com/erew123/alltalk_tts/tree/alltalkbeta/system/requirements Obviously there are other files installed as sub requirements of the main requirements. 

If you want a full list all all requirements installed in the AllTalk Python environment, you can run `start_diagnotics.xx` (for your OS) and generate a diagnostics.log file, In that file, it will list out all the current versions of everything installed into the AllTalk Python environment.

If you have any further questions, please let me know.

Thanks

**Reply by Enchante503** [reply]
Hello erew123
Thank you so much, your detailed explanation is very helpful.
I was having trouble with Cutlass.
